eurocodedesign/stepper/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""
+Thread-safe module for collecting printable information on calculation steps
+
+Examples
+    from eurocodedesign import stepper
+    >>> with stepper.create() as st1:
+    >>>    section = "IPE300"
+    >>>    st1.step(f"Steel section {section} has height 3.0 m.")
+    >>>    # do stuff
+    >>>    st1.step(f"{section} is of cross-section type {QK}.")
+    Steel section IPE300 has height 3.0 m. IPE300 is of cross-section type QK3.
+    >>> st2 = stepper.create()
+    >>> st2.step("Add step")
+    >>> del st2
+    Add step
+"""
+from collections import deque
+from typing import Self
+
+
+class Stepper:
+    """
+    Thread-safe context manager for collection calculation steps
+    """
+    _steps: deque
+
+    def __init__(self) -> None:
+        self._steps = deque()
+
+    def __str__(self) -> str:
+        """
+        Returns: Added steps concatenated to one string separated by spaces.
+        """
+        if self._steps:
```

Comment:
is this not always true given that `self._steps` is declared in `__init__`? I think you can remove the if statement and just have `return ' '.join(self._steps)` whilst retaining the same functionality